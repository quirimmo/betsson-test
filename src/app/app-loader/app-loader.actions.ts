import { Injectable } from '@angular/core';
import { Action } from 'redux';
import { NgRedux } from '@angular-redux/store';
import { StoreModel } from '../store/store.model';

export interface AppLoaderAction extends Action {
	isLoading: boolean;
}

@Injectable()
export class AppLoaderActions {
	static readonly START_LOADING = 'START_LOADING';
	static readonly STOP_LOADING = 'STOP_LOADING';

	constructor(private ngRedux: NgRedux<StoreModel>) {}

	public startLoading(): AppLoaderAction {
		return this.ngRedux.dispatch({
			type: AppLoaderActions.START_LOADING,
			isLoading: true
		});
	}

	public stopLoading(): AppLoaderAction {
		return this.ngRedux.dispatch({
			type: AppLoaderActions.STOP_LOADING,
			isLoading: false
		});
	}
}
