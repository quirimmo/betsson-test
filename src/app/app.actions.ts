import { Injectable } from '@angular/core';
import { Action } from 'redux';
import { NgRedux } from '@angular-redux/store';
import { StoreModel } from './store/store.model';

@Injectable()
export class AppActions {
	static readonly START_LOADING = 'START_LOADING';
	static readonly STOP_LOADING = 'STOP_LOADING';

	constructor(private ngRedux: NgRedux<StoreModel>) {}

	public startLoading(): Action {
		return this.ngRedux.dispatch({
			type: AppActions.START_LOADING
		});
	}

	public stopLoading(): Action {
		return this.ngRedux.dispatch({
			type: AppActions.STOP_LOADING
		});
	}
}
