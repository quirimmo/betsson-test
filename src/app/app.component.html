<div class="app-wrapper container-fluid">
	<div *ngIf="(isLoading$ | async); else loaded_content">
		THE APP IS LOADING
	</div>
	<ng-template #loaded_content>
		<movies-list [movies]="movies$ | async"> </movies-list>
	</ng-template>
</div>
