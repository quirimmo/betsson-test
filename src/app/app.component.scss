@import "./../colors.scss";
.app-wrapper {
  background-color: $black;
}
