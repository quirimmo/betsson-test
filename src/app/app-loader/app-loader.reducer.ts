import { AppLoaderAction, AppLoaderActions } from './app-loader.actions';

export function appLoaderReducer(
	state: boolean = false,
	action: AppLoaderAction
): boolean {
	switch (action.type) {
		case AppLoaderActions.START_LOADING:
			return action.isLoading;
		case AppLoaderActions.STOP_LOADING:
			return action.isLoading;
		default:
			return state;
	}
}
