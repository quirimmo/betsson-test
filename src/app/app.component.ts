import { Component, OnInit, AfterViewInit } from '@angular/core';
import { MoviesActions } from './movies/movies.actions';
import { select } from '@angular-redux/store';
import { Observable } from 'rxjs';
import Movie from './movies/movie.model';
import { AppLoaderActions } from './app-loader/app-loader.actions';

@Component({
	selector: 'app-root',
	templateUrl: './app.component.html',
	styleUrls: ['./app.component.scss']
})
export class AppComponent implements OnInit {
	title = 'betsoon-test';

	constructor(
		private moviesActions: MoviesActions,
		private appActions: AppLoaderActions
	) {}

	public ngOnInit(): void {}
}
