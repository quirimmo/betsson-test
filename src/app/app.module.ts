import { BrowserModule } from '@angular/platform-browser';
import { CommonModule } from '@angular/common';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NgReduxModule } from '@angular-redux/store';
import { NgReduxRouterModule } from '@angular-redux/router';
import { NgReduxFormModule } from '@angular-redux/form';
import { StoreModule } from './store/store.module';
import { AppComponent } from './app.component';
import { MoviesModule } from './movies/movies.module';
import { AppLoaderModule } from './app-loader/app-loader.module';
import { AppRoutes } from './app.routes';
import { RoutesModule } from './routes/routes.module';
import { ResolversModule } from './resolvers/resolvers.module';

@NgModule({
	declarations: [AppComponent],
	exports: [],
	imports: [
		CommonModule,
		BrowserModule,
		BrowserAnimationsModule,
		FormsModule,
		ReactiveFormsModule,
		RouterModule.forRoot(AppRoutes),
		NgReduxModule,
		NgReduxRouterModule.forRoot(),
		NgReduxFormModule,
		StoreModule,
		MoviesModule,
		AppLoaderModule,
		RoutesModule,
		ResolversModule
	],
	providers: [],
	bootstrap: [AppComponent]
})
export class AppModule {}
