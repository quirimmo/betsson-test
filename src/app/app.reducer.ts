import { AppActions } from './app.actions';
import { Action } from 'redux';

export function appReducer(state: boolean = false, action: Action): boolean {
	switch (action.type) {
		case AppActions.START_LOADING:
			return true;
		case AppActions.STOP_LOADING:
			return false;
		default:
			return state;
	}
}
