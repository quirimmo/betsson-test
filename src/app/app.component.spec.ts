import { TestBed, async } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { AppLoaderComponent } from './app-loader/app-loader.component';
import { RouterTestingModule } from '@angular/router/testing';

describe('AppComponent', () => {
	let appComponentFixture, appComponentInstance;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [AppComponent, AppLoaderComponent],
			imports: [RouterTestingModule]
		}).compileComponents();
		appComponentFixture = TestBed.createComponent(AppComponent);
		appComponentInstance = appComponentFixture.debugElement.componentInstance;
	}));

	it('should create the component', () => {
		expect(appComponentInstance).toBeTruthy();
	});

	it('should match the snapshot', () => {
		expect(appComponentFixture).toMatchSnapshot();
	});

	it(`should have as title 'betsoon-test'`, () => {
		expect(appComponentInstance.title).toEqual('betsoon-test');
	});
});
