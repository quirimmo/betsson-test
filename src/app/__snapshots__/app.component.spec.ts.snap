// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match the snapshot 1`] = `
<app-root
  title={[Function String]}
>
  <div
    class="app-wrapper"
  >
    <app-loader>
      
    </app-loader>
    <router-outlet />
  </div>
</app-root>
`;
